### DISCUSSION TITLE
How to append an id to a multiple record relation field

### DISCUSSION THREAD

**Discussion by hleinan** [pocketbase/pocketbase#291] [Q&A]
Hello. 

I have a tavle with a relation to another table. This relation is one to many. However, when updating (adding one relation), it looks like I have to send the alteady existing relations from the frontend, or the record is updated with only the newest relation. 

Is there a simple way to add one extra relation whitout adding all existing relations (i see that a race condition could easily happen if I need to send them all at every add)

Thanks. And great work btw

**Comment by ganigeorgiev** [comment]
Hi, unfortunately at the moment No - you can't append a value to an existing multivalue field (there is an exception only for the `file` type field, because file uploads are handled differently).

This could be useful, but I'm not sure how to implement it without introducing large breaking changes. Suggestions for implementation/data format are welcome.

One hacky workaround could be to allow specifying a suffixed special character (eg. `+`) at the end of the field name, that will indicate `append` type of operation, eg.:
```js
// sample json body
{
    "title": "test",
    "rel+": "YOUR_REL_ID",
    ...
}
```

> i see that a race condition could easily happen if I need to send them all at every add

This is true for every other field value when doing concurrent updates. PocketBase currently doesn't have an integrated mechanism for this and CRDT-like data operations are left to the users to handle on their own.

**Reply by ganigeorgiev** [reply]
_Just for info if anyone is following here, this is already supported in the latest [v0.11.0 release](https://github.com/pocketbase/pocketbase/releases/tag/v0.11.0)._
